### PR TITLE
[Azure.Core] Add Newton soft BYOM case

### LINKED ISSUE #36480: [Azure.Core] Change Serializer on SerializableOptions to be a dictionary for BYOM
for BYOM case:
check if what we're deserializing exists in dict, use that

### PR DESCRIPTION
**Pull request by nisha-bhatia**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/core/Azure.Core/src/Serialization/SerializableOptions.cs [hunk 1]
```diff
@@ -23,9 +26,14 @@ public class SerializableOptions
...
         public bool PrettyPrint { get; set; }
 
+        ///// <summary>
+        ///// todo
+        ///// </summary>
+        //public ObjectSerializer? Serializer { get; set; }
+
         /// <summary>
-        /// todo
+        /// TODO
```
Comment (christothes) [inline sdk/core/Azure.Core/src/Serialization/SerializableOptions.cs:35]: docs?

File: sdk/core/Azure.Core/src/Serialization/SerializableOptions.cs [hunk 2]
```diff
@@ -23,9 +26,14 @@ public class SerializableOptions
         /// </summary>
         public bool PrettyPrint { get; set; }
 
+        ///// <summary>
+        ///// todo
+        ///// </summary>
+        //public ObjectSerializer? Serializer { get; set; }
```
Comment (christothes) [inline sdk/core/Azure.Core/src/Serialization/SerializableOptions.cs:32]: remove?

File: sdk/core/Azure.Core/src/Serialization/SerializableOptions.cs [hunk 3]
```diff
@@ -23,9 +26,14 @@ public class SerializableOptions
...
+
         /// <summary>
-        /// todo
+        /// TODO
         /// </summary>
-        public ObjectSerializer? Serializer { get; set; }
+        public Dictionary<Type, ObjectSerializer> Serializers = new Dictionary<Type, ObjectSerializer>();
```
Comment (christothes) [inline sdk/core/Azure.Core/src/Serialization/SerializableOptions.cs:37]: Should we mark this as `readonly` or add  `{ get; }`?

Comment (nisha-bhatia) [reply sdk/core/Azure.Core/src/Serialization/SerializableOptions.cs:32]: I think this cannot be readonly as the user needs to be able to add model types and their associated serializers.

Comment (christothes) [reply sdk/core/Azure.Core/src/Serialization/SerializableOptions.cs:32]: They would still be able to add elements to the dictionary - just not replace the dictionary instance.

**Comment by KrzysztofCwalina** [comment]
@nisha-bhatia, could you add a description to this PR describing the motivation for this change?
Thanks.

File: sdk/core/Azure.Core/src/Serialization/SerializableOptions.cs (hunk 3 above)
Comment (nisha-bhatia) [reply sdk/core/Azure.Core/src/Serialization/SerializableOptions.cs:32]: sorry Chris I misunderstood you yesterday - updating now.

**Review by m-nash — APPROVED** [review]

**Pull request merged**